arangod/IResearch/IResearchLink.cpp

```diff
@@ -823,7 +835,12 @@ Result IResearchLink::commit(bool wait /*= true*/) {
   }
 
   CommitResult code{CommitResult::UNDEFINED};
-  return commitUnsafe(wait, &code).result;
+  auto result = commitUnsafe(wait, &code).result;
+  if (auto& meta = _dataStore._meta;
+      meta._commitIntervalMsec == 0 && meta._cleanupIntervalStep != 0) {
```

Comment:
In case of disabled periodic commits this will cleanup on each explicit commit and ignore configured cleanup step (1 cleanup per N commits). Is this intended? For production this is a very unlikely case, but nonetheless.

Reply:
Fixed